src/runtime/external.ml

```diff
@@ -84,19 +84,29 @@ let externals =
          Runtime.return_closure (fun der ->
              let chk = Runtime.as_equality_checker ~at:Location.unknown chk
              and drv = Runtime.as_derivation ~at:Location.unknown der in
-             match Eqchk.add_type_computation chk drv with
-             | Some chk -> Runtime.return (Reflect.mk_option (Some Runtime.(External (EqualityChecker chk))))
-             | None -> Runtime.return (Reflect.mk_option None)
+             try
```

Comment:
This `try .. with ` pattern is repeated 6 times ; should it be hoisted out as a wrapper function?